third_party/docfx-doclet-143274/src/main/java/com/microsoft/lookup/BaseLookup.java

```diff
@@ -313,4 +327,110 @@ private String getSeeTagRef(SeeTree seeTree) {
+    }
+    return null;
+  }
+
+  public String getStatusComment(T element) {
+    Map<String, Optional<String>> annotationComments = getAnnotationComments(element);
+
+    // Deprecated comments are determined by the Javadoc @deprecated block tag.
+    // See this#replaceBlockTags
+
+    List<String> comments = new ArrayList<>();
+    if (annotationComments.containsKey("InternalApi")) {
+      comments.add(createInternalOnlyNotice(annotationComments.get("InternalApi")));
+    }
+    if (annotationComments.containsKey("InternalExtensionOnly")) {
+      comments.add(
+          createInternalExtensionOnlyNotice(annotationComments.get("InternalExtensionOnly")));
+    }
+    if (annotationComments.containsKey("ObsoleteApi")) {
+      comments.add(createObsoleteNotice(annotationComments.get("ObsoleteApi")));
+    }
+    if (annotationComments.containsKey("BetaApi")) {
+      comments.add(createBetaNotice(annotationComments.get("BetaApi")));
+    }
+
+    if (comments.isEmpty()) {
+      return null;
+    }
+    return String.join("\n\n", comments);
+  }
+
+  private String createBetaNotice(Optional<String> customComment) {
+    return "<aside class=\"beta\">\n"
+        + "<p><strong>Beta</strong></p>\n"
+        + customComment.map(comment -> "<p><em>" + comment + "</em></p>\n").orElse("")
+        + "<p>This feature is covered by the <a href=\"/terms/service-terms#1\">Pre-GA Offerings "
+        + "Terms</a> of the Terms of Service. Pre-GA libraries might have limited support, and "
+        + "changes to pre-GA libraries might not be compatible with other pre-GA versions. For "
+        + "more information, see the launch stage descriptions.</p>\n"
+        + "</aside>\n";
+  }
+
+  private String createObsoleteNotice(Optional<String> customComment) {
+    return "<aside class=\"deprecated\">\n"
+        + "<p><strong>Obsolete</strong></p>\n"
+        + customComment.map(comment -> "<p><em>" + comment + "</em></p>\n").orElse("")
+        + "<p>This feature is stable for usage in this major version, but may be deprecated in a "
+        + "future release.</p>\n"
+        + "</aside>\n";
+  }
+
+  private String createInternalExtensionOnlyNotice(Optional<String> customComment) {
+    return "<aside class=\"special\">\n"
+        + "<p><strong>Internal Extension Only</strong>: This feature is stable for usage, but is "
+        + "not intended for extension or implementation.</p>\n"
+        + customComment.map(comment -> "<p><em>" + comment + "</em></p>\n").orElse("")
+        + "</aside>\n";
+  }
+
+  private String createInternalOnlyNotice(Optional<String> customComment) {
+    return "<aside class=\"warning\">\n"
+        + "<p><strong>Internal Only</strong>: This feature is not stable for application use.</p>\n"
+        + customComment.map(comment -> "<p><em>" + comment + "</em></p>\n").orElse("")
+        + "</aside>\n";
+  }
+
+  /**
+   * @return all annotations on the element and their associated comment, if it exists

```

Comment:
Am I reading this right such that any annotation, even not `InternalApi`, `Deprecated`, `ObsoleteApi`, or `BetaApi`, will still get surfaced ? 
If so, is it possible to get an example of what that would look like in staging? b/288904862 has an example for Storage.

Reply:
The CL has staging examples of methods with BetaAPI which is the storage request in that bug.

Staging also has examples of Internal API, but there aren't any current examples of Obsolete API and these updates don't change Deprecated behavior.

Reply:
Ah I thought the original request was to support the `@TransportCompatibility` annotation, which was then followed by a request to support the `@BetaApi` annotation. I think this code should support both, but wanted to doublecheck how a generic annotation would appear.

Reply:
Ah - I understand. Yes, you make a good point - the original request was about supporting the `@Documented` annotation which would generically support any annotation -- but after discussing with Ben we agreed to limit the scope to @Beta only. The fact that Internal, Internal Extension Only, and Obsolete are supported is due to a request by the team when discussing the agreement.

So `@Documented` support is currently out of scope of this PR.